### DISCUSSION TITLE
How to change MIDI CC of a modulator from Lua?

### DISCUSSION THREAD

**Discussion by dervur** [RomanKubiak/ctrlr#667] [Q&A] [answered]
Hello there, 

Is there a way to check and change CC values of modulators via Lua? I think it should be possible but I'm missing something here I think cause my code crashes.

The one to get the CC should be:
```
    local CC = 0
    CC = panel:getModulatorByName("Sample Part-1"):getProperty("midiMessageCtrlrNumber")
```
Right? This does not make Ctrlr crash.

However even doing `console(CC.."CC")` makes it crash so I don't understand if the way I'm getting the currently set CC value is wrong or what else is making the console string crash.

I can't seem to be able to set a CC either, what I'm doing is:
`panel:getModulatorByName("Sample Part-1"):setProperty("midiMessageCtrlrNumber", 5)`
This makes it crash as well.

Does anybody know if what I want to do is possible and if yes where is my mistake?

Cheers

**Comment by dervur** [comment]
Ok it appears that also my code above is doing nothing albeit it runs because `console(type(CC))` returns "nil", so this solves why `console(CC)` made Ctrlr crash so I can't seem to get the property either...

**Reply by dervur** [reply]
Other things I tried:
```
local mod = panel:getModulatorByName("Sample Part-1")
    local CC = mod:getMidiMessage():getProperty("midiMessageCtrlrNumber")
```
Makes it crash.
```
local mod = panel:getModulatorByName("Sample Part-1")
    local CC = mod:getComponent():getProperty("midiMessageCtrlrNumber")
```
CC is nil...

**Comment by dervur** [comment]
Oh I just saw https://github.com/RomanKubiak/ctrlr/discussions/619 so it is a known issue...
I'll leave the discussion open for future reference until this is fixed.

**Comment by dnaldoog** [comment, marked as answer]
The following works on 5.3.201, but crashes on my build of 5.6!
You had the _getProperty_ code correct but `SetProperty()` needs a [third](https://github.com/search?q=repo%3ARomanKubiak%2Fctrlr+setProperty&type=code) boolean parameter
<code>
panel:getModulatorByName("Sample Part-1"):getMidiMessage():setProperty("midiMessageCtrlrNumber", 5,true)
console("okay")


local mod = panel:getModulatorByName("Sample Part-1")
    local CC = mod:getMidiMessage():getProperty("midiMessageCtrlrNumber")
console(CC.."CC")
</code>

**Reply by dervur** [reply]
Thank you very much! Yeah I saw your comment on https://github.com/RomanKubiak/ctrlr/discussions/619 after posting this question... Thanks for the correction on the `setProperty()`!
